dali/operators/random/choice_cpu.cc

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/random/choice.h"
+#include "dali/operators/random/rng_base_cpu.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/operator/arg_helper.h"
+
+namespace dali {
+
+DALI_SCHEMA(random__Choice)
+    .DocStr(R"code(Generates a random sample from a given 1D array.
```

Comment:
Not only: _if a scalar is provided, values from [0, a) are sampled_

Reply:
I adjusted the doc for the first input, so this now holds for all inputs.